APITaxi2/views/taxis.py

```diff
@@ -230,12 +230,22 @@ def taxis_details(taxi_id):
         }, status_code=404)
     taxi, vehicle_description = (res.Taxi, res.VehicleDescription)
 
+    redis_taxi = redis_backend.get_taxi(taxi.id, vehicle_description.added_by.email)
+    location = None
+    if redis_taxi:
+        location = redis_backend.Location(
+            lon=redis_taxi.lon,
+            lat=redis_taxi.lat,
+            distance=None,
+            update_date=datetime.fromtimestamp(redis_taxi.timestamp)
+        )
```

Comment:
Correct me if I'm wrong, but following the example in your PR description, you're still returning the outdated location2, since you're filtering by the operator2 email, not the latest known position.

Reply:
Indeed, but it is expected, this is GET /taxis/:id and not GET /taxis.

`GET /taxis/:id` is an operator endpoint, and you expect to retrieve the last location of your operator.
`GET /taxis?lon=xx&lat=yy` is a search provider endpoint, where you expect to get a fresh location.